### ISSUE TITLE
Default type parameters on struct give spurious error when deriving

### ISSUE THREAD

**Issue by bonsairobo** [djkoloski/rkyv#82]
I'm on version 0.4.

Example:

```rust
#[derive(rkyv::Archive, rkyv::Deserialize, rkyv::Serialize)]
struct Foo<T = i32> {
    t: T,
}
```

Produces:

```
defaults for type parameters are only allowed in `struct`, `enum`, `type`, or `trait` definitions.
```

**Comment by djkoloski** [comment]
The derive macro is getting cleaned up for 0.5 and I'll make sure this is one of the test cases it passes.

If you need a fix sooner in 0.4.x, I'll accept a pull request that fixes it.

**Comment by bonsairobo** [comment]
No rush. Thanks for the reply.

**Comment by djkoloski** [comment]
Fixed by 0e0bd4ecc5e7a2ee0fa5156552c877883b816723, 000347f003b1cc062e8b01b6695d67afe26f97fb, daf853ac34bf066d72053e892daf7e1800fc8955. 0.5 will have the fix when it's released; you can get it now using:

```toml
[dependencies]
rkyv = { git = "https://github.com/djkoloski/rkyv", branch = "0.5.0" }
```
